sidecar/internal/service/reclaimspace.go

```diff
@@ -88,14 +110,34 @@ func (rs *ReclaimSpaceServer) NodeReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	var deviceSecret map[string]string
+	var volID string
+	if pv.Spec.CSI == nil {
+		volID = pv.Spec.CSI.VolumeHandle
+		if pv.Spec.CSI.NodeStageSecretRef != nil {
+			// Get the secrets from the k8s cluster
+			deviceSecret, err = kube.GetSecret(ctx, rs.kubeClient, pv.Spec.CSI.NodeStageSecretRef.Name, pv.Spec.CSI.NodeStageSecretRef.Namespace)
+			if err != nil {
+				klog.Errorf("Failed to get secret %v", err)
+				return nil, status.Error(codes.InvalidArgument, err.Error())
+			}
+		}
```

Comment:
move this to a helper function and reuse it?

Reply:
it is slightly different for both cases.  In the case of controllerreclaimspace we need to get volume attributes while for node we don't need that. Hence, in my opinion, it is fine.